### PR TITLE
:telephone_receiver:  IPC & multi threading communication

### PR DESCRIPTION
**Pull request by landryarki**
see:
- #8

I added 2 messages queue classes, one for threads communication and one for processes communication.

### CODE REVIEW

File: include/messageQueueThread.hpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            return T();
+        }
+        return _queue.front();
+    }
+
+    /**
+     * @brief Checks if the message queue is empty.
+     * @return True if the queue is empty, false otherwise.
+     */
+    bool empty()
```
Comment (Peralban) [inline include/messageQueueThread.hpp:74]: bool empty in cpp

File: include/messageQueueThread.hpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    /**
+     * @brief Default destructor for the MessageQueueThread class.
+     */
+    ~MessageQueueThread() = default;
+
+    /**
+     * @brief Pushes a message to the message queue.
+     * @param message The message to be pushed.
+     */
+    void push(T message)
```
Comment (Peralban) [inline include/messageQueueThread.hpp:36]: void push in cpp

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
You should put the methods code of the `messageQueueIPC.hpp` file in a cpp one :wink:

File: include/messageQueueIPC.hpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    /**
+     * @brief Checks if the message queue is empty.
+     * @return True if the queue is empty, false otherwise.
+     */
+    bool empty()
+    {
+        message_buf buf = {};
+        msgrcv(_msg_id, &buf, sizeof(buf.mtext), 0, MSG_COPY | IPC_NOWAIT);
+        return buf.mtext[0] == 0;
+    }
```
Comment (Peralban) [inline include/messageQueueIPC.hpp:81]: All of this in cpp

**Review by Peralban — CHANGES_REQUESTED** [review]
You can put functions in cpp file

**Review by Thomaltarix — APPROVED** [review]
Clean one

**Pull request merged**